src/components/taskDetails/Details.tsx

```diff
@@ -1,20 +1,86 @@
-import React, { FC, useState } from 'react';
+import React, { FC, useEffect, useState } from 'react';
 import moment from 'moment';
 import { FaReceipt } from 'react-icons/fa6';
 import Link from 'next/link';
 import Tooltip from '@/components/common/Tooltip/Tooltip';
 import setColor from './taskPriorityColors';
 import extractRepoName from '@/utils/extractRepoName';
 import styles from './task-details.module.scss';
-import { TaskDetailsProps } from '@/interfaces/taskDetails.type';
+import {
+    DetailsContentProps,
+    TaskDetailsProps,
+} from '@/interfaces/taskDetails.type';
+import useUserData from '@/hooks/useUserData';
 import { STARTED_ON, ENDS_ON } from '@/constants/constants';
 
 type StringNumberOrUndefined = string | number | undefined;
 
-const Details: FC<TaskDetailsProps> = ({ detailType, value, url }) => {
+const DetailsContent: FC<DetailsContentProps> = ({
+    color,
+    isGitHubLink,
+    value,
+    gitHubIssueLink,
+    isTimeDetail,
+    formatDate,
+    tooltipActive,
+    renderedValue,
+    getRelativeTime,
+}) => {
+    return (
+        <span
+            className={styles.detailValue}
+            style={{ color: color ?? 'black' }}
+        >
+            {isGitHubLink && value ? (
+                <a
+                    className={styles.gitLink}
+                    href={gitHubIssueLink}
+                    target="_blank"
+                    rel="noopener noreferrer"
+                    aria-label="Open GitHub Issue"
+                    title={value}
+                >
+                    {isGitHubLink ? `${extractRepoName(value)}` : value}
+                </a>
+            ) : isTimeDetail ? (
+                <Tooltip
+                    content={formatDate(value)}
+                    tooltipPosition={{
+                        top: '-3.6rem',
+                        right: '-4.5rem',
+                    }}
+                >
+                    {tooltipActive ? formatDate(value) : getRelativeTime(value)}
+                </Tooltip>
+            ) : (
+                renderedValue
+            )}
+        </span>
+    );
+};
+
+const Details: FC<TaskDetailsProps> = (props) => {
+    const { detailType, value, url, isEditing, setEditedTaskDetails } = props;
     const color = value ? setColor?.[value] : undefined;
     const isGitHubLink = detailType === 'Link';
     const gitHubIssueLink = isGitHubLink ? value : undefined;
+    const [newEndOnDate, setNewEndOnDate] = useState('');
+    const { isUserAuthorized } = useUserData();
+
+    useEffect(() => {
+        if (!isEditing && newEndOnDate !== '') setNewEndOnDate('');
+    }, [isEditing, newEndOnDate]);
+
+    const handleBlurOfEndsOn = () => {
+        const endsOn = new Date(`${newEndOnDate}`).getTime() / 1000;
```

Comment:
usage of `new Date()` can lead to unexpected results with invalid date formats, as `new Date()` will convert invalid dates to `NaN`. This will likely prevent proper error handling for invalid dates.

use a more robust date validation method (e.g., `Date.parse()` or a library like `date-fns`) to ensure that invalid dates are handled gracefully.

Reply:
Ok will check this

Reply:
fixed